Infoblox-CDC-Solutions-queries

sschuur opened the pull request:
Fixes #

## Proposed Changes

  - uploaded analytics rules

shainw commented on Solutions/Infoblox Cloud Data Connector/Analytic Rules/HighNumberOfHighThreatLevelDetected.yaml:
```diff
@@ -0,0 +1,41 @@
+  - T1565
+query: |
+
+  let timeframe = 1h;
```
This can be removed as it is not used and the queryPeriod will be used for the lookback timeframe

shainw commented on Solutions/Infoblox Cloud Data Connector/Analytic Rules/HighNumberofNXDOMAINDNSQueriesDetected.yaml:
```diff
@@ -0,0 +1,39 @@
+  - T1565
+query: |
+
+  let timeframe = 1h;
```
timeframe can be removed.

shainw commented on Solutions/Infoblox Cloud Data Connector/Analytic Rules/HighNumberOfHighThreatLevelDetected.yaml:
```diff
@@ -0,0 +1,41 @@
+  let threshold = 200;
+  CommonSecurityLog
+  | where DeviceVendor == "Infoblox" and DeviceProduct == "Data Connector"
+  | where DeviceEventClassID contains "RPZ"
```
If possible, use ```has``` instead of ```contains```.  If it is the start or end of a string you can also use ```hasprefix``` or ```hassuffix```.

shainw commented on Solutions/Infoblox Cloud Data Connector/Analytic Rules/HighNumberOfHighThreatLevelDetected.yaml:
```diff
@@ -0,0 +1,41 @@
+  | where count_ > threshold
+  | join kind=inner (CommonSecurityLog
+      | where DeviceVendor == "Infoblox" and DeviceProduct == "Data Connector"
+      | where DeviceEventClassID contains "RPZ"
```
If possible, use ```has``` instead of ```contains```.  If it is the start or end of a string you can also use ```hasprefix``` or ```hassuffix```.

shainw commented on Solutions/Infoblox Cloud Data Connector/Analytic Rules/HighNumberofNXDOMAINDNSQueriesDetected.yaml:
```diff
@@ -0,0 +1,39 @@
+  let threshold = 200;
+  CommonSecurityLog
+  | where DeviceVendor == "Infoblox" and DeviceProduct == "Data Connector"
+  | where DeviceEventClassID contains "RPZ"
```
If possible, use ```has``` instead of ```contains```.  If it is the start or end of a string you can also use ```hasprefix``` or ```hassuffix```.

shainw commented on Solutions/Infoblox Cloud Data Connector/Analytic Rules/HighNumberofNXDOMAINDNSQueriesDetected.yaml:
```diff
@@ -0,0 +1,39 @@
+  | where count_ > threshold
+  | join kind=inner (CommonSecurityLog
+      | where DeviceVendor == "Infoblox" and DeviceProduct == "Data Connector"
+      | where DeviceEventClassID contains "RPZ"
```
If possible, use ```has``` instead of ```contains```.  If it is the start or end of a string you can also use ```hasprefix``` or ```hassuffix```.

shainw reviewed (changes requested):
Some minor changes.  Question - It looks like "High Number of High Threat Level Detected" will include results from "High Number of NXDOMAIN DNS Queries Detected" since the second just adds this filter ```| where SimplifiedDeviceAction == "NXDOMAIN"```.  Should you add an exclusion filter for this same thing to "High Number of High Threat Level Detected"?

sschuur commented:
Hi @shainw, I changed all my contains to has and removed the unused timeframe lines.  
I left them able to overlap since high threat levels can also be nxdomains. So regardless of the SimplifiedDeviceAction, a high threat level will be indicative. Do you think they should be mutually exclusive?

shainw reviewed (approved):
@sschuur - All good, we can leave them as be, you are the expert here, I was only wondering if that was on purpose and clearly it is. :)
